File: java/smartlist/src/main/java/com/smartlist/module/assessment/dto/response/AssessmentRespDTO.java
package com.smartlist.module.assessment.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.smartlist.model.Assessment;
import com.smartlist.model.CandidateResponse;
import com.smartlist.module.candidate.dto.response.CandidateResponseRespDTO;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Getter;
import lombok.Setter;
import org.modelmapper.Converter;
import org.modelmapper.ModelMapper;
import org.modelmapper.TypeMap;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class AssessmentRespDTO {

    @Schema(description = "Assessment Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    Integer id;

    @Schema(description = "Assessment Name", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    String name;

    @Schema(description = "Job Position", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    String position;

    @Schema(description = "Department", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    String department;

    @Schema(description = "Assessment Introduction", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    String introduction;

    @Schema(description = "Assessment Status", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    String status;

    @Schema(description = "Assessment Token", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotBlank
    String token;

    @Schema(description = "Header Image", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "header_image_id")
    Integer headerImageId;

    @Schema(description = "Video Link URL", example = "", requiredMode= Schema.RequiredMode.NOT_REQUIRED)
    @JsonProperty(value = "video_link_url")
    String videoLinkUrl;

    @Schema(description = "Organization Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @NotNull
    @Positive
    @JsonProperty(value = "organization_id")
    Integer organizationId;


}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/assessment/dto/response/assessment.resp.dto.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { IsInt, IsNotEmpty, IsOptional, IsUrl, Min } from 'class-validator';
import { Assessment } from 'src/entities/assessment.entity';

export class AssessmentRespDto {
  @ApiProperty({
    type: Number,
  })
  @IsNotEmpty()
  id: number;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  name: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  position: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  department: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  introduction: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  status: string;

  @ApiProperty({
    type: String,
  })
  @IsNotEmpty()
  token: string;

  @ApiPropertyOptional({
    type: Number,
  })
  @IsInt()
  @Min(1)
  header_image_id: number;

  @ApiPropertyOptional({
    type: String,
  })
  @IsOptional()
  @IsUrl()
  video_link_url: string;

  @ApiProperty({
    type: Number,
  })
  @IsInt()
  @Min(1)
  organization_id: number;

  constructor(assessment: Assessment) {
    this.id = assessment.id;
    this.name = assessment.name;
    this.position = assessment.position;
    this.department = assessment.department;
    this.introduction = assessment.introduction;
    this.status = assessment.status;
    this.token = assessment.token;
    this.header_image_id =
      assessment.headerImage != null ? assessment.headerImage.id : null;
    this.video_link_url = assessment.videoLinkURL;
    this.organization_id =
      assessment.organization != null ? assessment.organization.id : null;
  }
}
